ui/components/MesheryCatalog.js

```diff
@@ -0,0 +1,1093 @@
+            { credentials: 'include', method: 'DELETE', body: JSON.stringify({ id: pattern?.id }) },
+            () => {
+              updateProgress({ showProgress: false });
+              notify({
+                message: `Design Unpublished`,
+                event_type: EVENT_TYPES.SUCCESS,
+              });
+            },
+            handleError(ACTION_TYPES.UNPUBLISH_CATALOG),
+          );
+        }
+      };
+    }
+  };
+
+  const handleDeploy = (pattern_file, pattern_id, name) => {
+    updateProgress({ showProgress: true });
+    dataFetch(
+      ctxUrl(DEPLOY_URL, selectedK8sContexts),
+      {
+        credentials: 'include',
+        method: 'POST',
+        body: JSON.stringify({
+          pattern_file: pattern_file,
+          pattern_id: pattern_id,
+        }),
+      },
+      () => {
+        updateProgress({ showProgress: false });
+        notify({
+          message: `"${name}" Design Deployed`,
+          event_type: EVENT_TYPES.SUCCESS,
+        });
+      },
+      handleError(ACTION_TYPES.DEPLOY_PATTERN),
+    );
+  };
+
+  const handleUnDeploy = (pattern_file, pattern_id, name) => {
+    updateProgress({ showProgress: true });
+    dataFetch(
+      ctxUrl(DEPLOY_URL, selectedK8sContexts),
+      {
+        credentials: 'include',
+        method: 'DELETE',
+        body: JSON.stringify({
+          pattern_file: pattern_file,
+          pattern_id: pattern_id,
+        }),
+      },
+      () => {
+        updateProgress({ showProgress: false });
+        notify({
+          message: `"${name}" Design undeployed`,
+          event_type: EVENT_TYPES.SUCCESS,
+        });
+      },
+      handleError(ACTION_TYPES.UNDEPLOY_PATTERN),
+    );
+  };
+
+  function handleClone(patternID, name) {
+    updateProgress({ showProgress: true });
+    dataFetch(
+      PATTERN_URL.concat(CLONE_URL, '/', patternID),
+      {
+        credentials: 'include',
+        method: 'POST',
+        body: JSON.stringify({ name: name + ' (Copy)' }),
+      },
+      () => {
+        updateProgress({ showProgress: false });
+        notify({
+          message: `"${name}" Design cloned`,
+          event_type: EVENT_TYPES.SUCCESS,
+        });
+      },
+      handleError(ACTION_TYPES.CLONE_PATTERN),
+    );
+  }
+
+  function fetchPatterns(page, pageSize, search, sortOrder, visibilityFilter) {
+    if (!search) search = '';
+    if (!sortOrder) sortOrder = '';
+    const query =
+      `?page=${page}&pagesize=${pageSize}&search=${encodeURIComponent(
+        search,
+      )}&order=${encodeURIComponent(sortOrder)}` +
+      (visibilityFilter
+        ? `&visibility=${encodeURIComponent(JSON.stringify([visibilityFilter]))}`
+        : '');
+
+    updateProgress({ showProgress: true });
+    dataFetch(
+      `/api/pattern${query}`,
+      { credentials: 'include' },
+      (result) => {
+        updateProgress({ showProgress: false });
+        if (result) {
+          const filteredPatterns = result.patterns.filter((content) => {
```

Comment:
no need to filter the api response , use published data directly from response . As you taken this code from designs page so there it was used to filter between visibility

Reply:
Sorry for not mentioning, After yesterday's team-meet:
- This whole component will be removed in favour of modifying and reusing the `MesheryPatterns` component used in the designs page.